dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/utils/DagHelper.java

```diff
@@ -565,20 +554,40 @@ public static boolean haveConditionsAfterNode(String parentNodeCode, List<TaskNo
         return false;
     }
 
+
     /**
      * is there have blocking node after the parent node
      */
     public static boolean haveBlockingAfterNode(String parentNodeCode,
                                                 DAG<String,TaskNode,TaskNodeRelation> dag) {
+        return haveSubAfterNode(parentNodeCode, dag, true, TaskConstants.TASK_TYPE_BLOCKING);
+    }
+
+    /**
+     * is there have all node after the parent node
+     */
+    public static boolean haveAllNodeAfterNode(String parentNodeCode,
+                                               DAG<String,TaskNode,TaskNodeRelation> dag) {
+        return haveSubAfterNode(parentNodeCode, dag, false, null);
+    }
+
+    /**
+     * Whether there is a specified type of child node after the parent node
+     */
+    public static boolean haveSubAfterNode(String parentNodeCode,
+                                           DAG<String,TaskNode,TaskNodeRelation> dag, boolean additionalNodeTypeFiltering, String filterNodeType) {
         Set<String> subsequentNodes = dag.getSubsequentNodes(parentNodeCode);
         if (CollectionUtils.isEmpty(subsequentNodes)) {
             return false;
         }
         for (String nodeName : subsequentNodes) {
             TaskNode taskNode = dag.getNode(nodeName);
             List<String> preTaskList = JSONUtils.toList(taskNode.getPreTasks(),String.class);
-            if (preTaskList.contains(parentNodeCode) && taskNode.isBlockingTask()) {
-                return true;
+            boolean containsParentNodeCode = preTaskList.contains(parentNodeCode);
+            if (additionalNodeTypeFiltering){
+                return containsParentNodeCode && taskNode.getType().equalsIgnoreCase(filterNodeType);
+            } else {
+                return containsParentNodeCode;
             }
         }
         return false;
```

Comment:
The logic is wrong in your change. In my knowledge, this method is used to check if there exist any subNode(filter by type) of the given parentNode.
See the below case, the input is `parentNodeA` and `Type2 `, the return result might be false, but the expect result is true.
<img width="465" alt="image" src="https://user-images.githubusercontent.com/22415594/164983262-d3b2232a-f4c1-4376-8a2a-3bdcd9b81e5f.png">

BTY, the parameter `additionalNodeTypeFiltering` `filterNodeType` is duplicate, you can just hold `filterNodeType` is enough.

Reply:
Yes, an error may occur in the case of multiple child nodes, a revision has been committed.